### DISCUSSION TITLE
Preload shaka player or other media ahead of time

### DISCUSSION THREAD

**Discussion by austinh** [muxinc/media-chrome#977] [Q&A]
Hello, i have an app with a grid of videos and I would like to preload them as the user scrolls, and stop loading them as the user scrolls past them (user unlikely to play those they scroll past).

shaka player provides a [preload](https://shaka-player-demo.appspot.com/docs/api/tutorial-preload.html) api to prefrtch the manifest and the first segment, as well as destroy/stop preloads midflight. Does this library expose any way for the media elements to preload things, or at least have access to the underlying instance?

hls.js also has a loadSource function, can I utilize that with hls-video?

if not, how do you recommend someone preload something (ideally from a cache)

**Comment by luwes** [comment]
related https://github.com/muxinc/media-elements/issues/37
